src/core/manager.c

```diff
@@ -1936,12 +1936,16 @@ static void manager_preset_all(Manager *m) {
         UnitFilePresetMode mode =
                 ENABLE_FIRST_BOOT_FULL_PRESET ? UNIT_FILE_PRESET_FULL : UNIT_FILE_PRESET_ENABLE_ONLY;
 
-        r = unit_file_preset_all(RUNTIME_SCOPE_SYSTEM, 0, NULL, mode, NULL, 0);
-        if (r < 0)
-                log_full_errno(r == -EEXIST ? LOG_NOTICE : LOG_WARNING, r,
-                               "Failed to populate /etc with preset unit settings, ignoring: %m");
-        else
-                log_info("Populated /etc with preset unit settings.");
+        RuntimeScope scope;
+
+        FOREACH_ARGUMENT(scope, RUNTIME_SCOPE_SYSTEM, RUNTIME_SCOPE_GLOBAL) {
+                r = unit_file_preset_all(scope, 0, NULL, mode, NULL, 0);
+                if (r < 0)
+                        log_full_errno(r == -EEXIST ? LOG_NOTICE : LOG_WARNING, r,
+                                       "Failed to populate /etc with %s preset unit settings, ignoring: %m", runtime_scope_to_string(scope));
+                else
+                        log_info("Populated /etc with %s preset unit settings.", runtime_scope_to_string(scope));
+        }
```

Comment:
uh.

this doesn't seem right to me. if we apply the per-user presets globally, then this means user's cannot cleanly disable them anymore, they would have to mask them.

Also, conceptually I doubt we should "reach over" from the system world to the user world.

If presets shall be applied automatically at some time, then we should probably do so in *user* scope, i.e. from systemd-user itself. Mabye in case the `~/.config/systemd` dir doesn#t exist (and then create it), or so. i.e. mirror the concept in a "first-login" way or so.